spec/vast_trackers.spec.js

```diff
@@ -282,5 +285,41 @@ describe('VASTTracker', function() {
         expect(spyTrack).not.toHaveBeenCalledTimes(2);
       });
     });
+    describe('#error', () => {
+      let isCustomCode = false;
+      beforeEach(() => {
+        vastTracker.error();
+      });
+      it('should be defined', () => {
+        expect(ad.errorURLTemplates).toBeDefined();
+      });
+      it('should have emitted an ad Error and called trackUrl ', () => {
+        expect(spyTrackUrl).toHaveBeenCalledWith(
+          ['http://example.com/error_[ERRORCODE]'],
+          expect.objectContaining(expectedMacros),
+          expect.objectContaining({ isCustomCode })
+        );
+        expect(spyTrackUrl).toHaveBeenCalledTimes(1);
+      });
+    });
+    describe('#errorWithCode', () => {
+      beforeEach(() => {
+        vastTracker.errorWithCode();
+      });
+      it('should be defined', () => {
+        expect(ad.errorURLTemplates).toBeDefined();
+      });
+      it('should hava called the error method', () => {
+        expect(spyAdError).toHaveBeenCalledTimes(1);
+      });
+      it('should have emitted an ad error, called trackUrl and not emitted any other event', () => {
+        expect(spyTrackUrl).toHaveBeenCalledWith(
```

Comment:
TrackUrl is not called inside the `errorWithCode` function so IMO we should not check that because if so it's more a end to end test. If we check that error is called with the right args it should be enough

Reply:
I agree